pallet/pallet-codegen/src/test/java/com/strategyobject/substrateclient/pallet/PalletInterfaceProcessorTests.java

```diff
@@ -0,0 +1,119 @@
+package com.strategyobject.substrateclient.pallet;
+
+import com.google.testing.compile.Compilation;
+import com.google.testing.compile.JavaFileObjects;
+import lombok.val;
+import org.junit.jupiter.api.Test;
+
+import static com.google.testing.compile.CompilationSubject.assertThat;
+import static com.google.testing.compile.Compiler.javac;
+
+public class PalletInterfaceProcessorTests {
+    @Test
+    void failsWhenAnnotationIsAppliedToClass() {
+        val clazz = JavaFileObjects.forResource("ClassPallet.java");
+
+        val compilation = javac()
+                .withProcessors(new PalletInterfaceProcessor())
+                .compile(clazz);
+
+        assertThat(compilation).failed();
+        assertThat(compilation)
+                .hadErrorContaining("Only interfaces");
+    }
+
+    @Test
+    void failsWhenPalletHasEmptyName() {
+        val clazz = JavaFileObjects.forResource("UnnamedPallet.java");
+
+        val compilation = javac()
+                .withProcessors(new PalletInterfaceProcessor())
+                .compile(clazz);
+
+        assertThat(compilation).failed();
+        assertThat(compilation)
+                .hadErrorContaining("contains null or empty `value`");
+    }
+
+    @Test
+    void failsWhenStorageHasEmptyName() {
+        val clazz = JavaFileObjects.forResource("UnnamedStorage.java");
+
+        val compilation = javac()
+                .withProcessors(new PalletInterfaceProcessor())
+                .compile(clazz);
+
+        assertThat(compilation).failed();
+        assertThat(compilation)
+                .hadErrorContaining("doesn't have `value`");
+    }
+
+    @Test
+    void failsWhenScaleIsNotSet() {
+        val clazz = JavaFileObjects.forResource("WithoutScale.java");
+
+        val compilation = javac()
+                .withProcessors(new PalletInterfaceProcessor())
+                .compile(clazz);
+
+        assertThat(compilation).failed();
+        assertThat(compilation)
+                .hadErrorContaining("Must be set `type` or `generic` of `@StorageKey`");
+    }
+
+    @Test
+    void failsWhenScaleTypeOfKeyIsAmbiguous() {
+        val clazz = JavaFileObjects.forResource("AmbiguousScale.java");
+
+        val compilation = javac()
+                .withProcessors(new PalletInterfaceProcessor())
+                .compile(clazz);
+
+        assertThat(compilation).failed();
+        assertThat(compilation)
+                .hadErrorContaining("Must be set only one of both parameters of `@StorageKey`: `type` or `generic`.");
```

Comment:
.hadErrorContaining("Please set only one parameter of `@StorageKey`: `type` or `generic`.");